src/emulate.c

```diff
@@ -277,17 +277,9 @@ void rv_debug(riscv_t *rv)
 }
 #endif /* RV32_HAS(GDBSTUB) */
 
-/* hash function is used when mapping address into the block map */
-static inline uint32_t hash(size_t k)
-{
-    k ^= k << 21;
-    k ^= k >> 17;
-#if (SIZE_MAX > 0xFFFFFFFF)
-    k ^= k >> 35;
-    k ^= k >> 51;
-#endif
-    return k;
-}
+#define HASH(val)                                                     \
```

Comment:
Could you create a macro that generates a hash function based on specific inputs like `cache_size_bits` and `BLOCK_MAP_CAPACITY_BITS`? By doing so, we can consolidate the `HASH` macros used in both `src/cache.c` and `src/emulate.c`. Furthermore, because the generated function body is in the form of a static inline function, the compiler can perform type checks.

Reply:
I make this modification, but it cause the performance lost. For example, the performance lost about 40 %(0.13s -> 0.22 s) when running the aes.elf. Due to the performance impact, I think marco is better even if it is not clean.

Reply:
How about using `FORCE_INLINE`?

Reply:
Ok, it works.